src/main/kotlin/com/openlattice/indexing/BackgroundLinkingIndexingService.kt

```diff
@@ -182,45 +176,30 @@ class BackgroundLinkingIndexingService(
     /**
      * Returns the linking ids, which are needing to be indexed.
      */
-    private fun getDirtyLinkingIds(): PostgresIterable<Pair<UUID, OffsetDateTime>> {
-        return PostgresIterable(Supplier<StatementHolder> {
-            val connection = hds.connection
-            val stmt = connection.prepareStatement(selectDirtyLinkingIds())
-            val rs = stmt.executeQuery()
-            StatementHolder(connection, stmt, rs)
-        }, Function<ResultSet, Pair<UUID, OffsetDateTime>> {
-            ResultSetAdapters.linkingId(it) to ResultSetAdapters.lastWriteTyped(it)
-        })
+    private fun getDirtyLinkingIds(): BasePostgresIterable<Pair<UUID, OffsetDateTime>> {
+        return BasePostgresIterable(
+                StatementHolderSupplier(hds, selectDirtyLinkingIds(), FETCH_SIZE)
+        ) { rs -> ResultSetAdapters.linkingId(rs) to ResultSetAdapters.lastWriteTyped(rs) }
     }
 
     private fun selectDirtyLinkingIds(): String {
         return "SELECT ${LINKING_ID.name}, ${DataTables.LAST_WRITE.name} " +
                 "FROM ${IDS.name} " +
                 "WHERE ${LINKING_ID.name} IS NOT NULL " +
-                    "AND ${LAST_INDEX.name} >= ${LAST_WRITE.name} " +
-                    "AND ${LAST_LINK.name} >= ${LAST_WRITE.name} " +
-                    "AND ${LAST_LINK_INDEX.name} < ${LAST_WRITE.name} " +
-                    "AND ${VERSION.name} > 0 " +
-                    "AND ${LINKING_ID.name} IS NOT NULL " +
-                "LIMIT $FETCH_SIZE"
+                "AND ${LAST_INDEX.name} >= ${LAST_WRITE.name} " +
+                "AND ${LAST_LINK.name} >= ${LAST_WRITE.name} " +
+                "AND ${LAST_LINK_INDEX.name} < ${LAST_WRITE.name} " +
+                "AND ${VERSION.name} > 0 AND ${LINKING_ID.name} IS NOT NULL"
```

Comment:
looks like `${LINKING_ID.name} IS NOT NULL` is already included in the where clause

Reply:
Nice catch! Yeah, just used autoformat on this, didn't really looked at the query, looks like second one got added with version > 0